### DISCUSSION TITLE
What is the right way to delploy nuxt3 to aws-lambda ?

### DISCUSSION THREAD

**Discussion by jerryisbusy** [nuxt/nuxt#16904] [Questions]
According to the official documentation, I tried to build my demo project like this:
`NITRO_PRESET=aws-lambda npm run build`, then i packed the .output folder to a zip file`zip output.zip -r .output/`.

And then, use `aws-lambda create-function` to create a function with the output.zip file by specify the handler as .output/server/index.handler.

Everything is ok by now,  after i attached an API gateway with my function, i could see the NuxtWelcome page rendered(Though, without css styles cause aws lambda does not support serving static files ).

**BUT!**

After i modified the `app.vue file`, replaced `<NuxtWelcome>` to `<NuxtPage>`, and then added `pages/index.vue`, all things got wrong.

All i have now is : {"message":"Internal Server Error"}.

My code works well  locally, i can't figure out what i did making things wrong.

Below is my API gateway url associated with my function:

[https://o29ahr3uug.execute-api.ap-east-1.amazonaws.com/default/nuxt3-demo](url)

Does some body know why ?

**Comment by jerryisbusy** [comment]
[demo.zip](https://github.com/nuxt/framework/files/8541237/demo.zip)

My demo project is Here.
It is almost a default project, the only change is added a pages/index.vue.
I created a Makefile to simplify my work, eg: create lambda function and update lambda function code.

**Comment by jerryisbusy** [comment]
Any body can help ?

**Comment by steffenstolze** [comment]
@jerryisbusy did you manage to find a solution hosting a SPA with API on AWS?
